pkg/render/json.go

```diff
@@ -38,6 +38,21 @@ import (
 // The buffers are fetched via a sync.Pool to reduce allocations and improve
 // performance.
 func (r *Renderer) RenderJSON(w http.ResponseWriter, code int, data interface{}) {
+	// Hello there reader! If you've made it here, you're likely wondering why
+	// you're getting an error about response codes. For client-interop, it's very
+	// important that we retain and maintain the allowed list of response codes.
+	// Adding a new response code requires coordination with the client team so
+	// they can update their applications to handle that new response code.
+	if !r.AllowedResponseCode(code) {
```

Comment:
naming nit: perhaps "AppHandledResponseCode" communicates the concept more directly? We don't want to send these codes back until client reference app handles them.